### PR TITLE
added JDBC

### PR DESCRIPTION
**Pull request by mikhail771**

### CODE REVIEW

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        } catch (SQLException e) {
+            throw new DataProcessingException("Product with id = " + id + " not found", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE product SET name = ?, "
+                + "price = ? WHERE product_id = ?;";
```
Comment (AnaLanda) [inline src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]:
```suggestion
                + "price = ? WHERE product_id = ? AND isDeleted = false;";
```

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            }
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Product " + product.getName() + " is not added!", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> getById(Long id) {
+        String query = "SELECT * FROM product WHERE product_id = ? AND isDeleted = FALSE;";
```
Comment (AnaLanda) [inline src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]:
```suggestion
        String query = "SELECT * FROM product WHERE product_id = ? AND isDeleted = false;";
```

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            statement.executeUpdate();
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Product " + product + " can't be updated", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE product SET isDeleted = TRUE WHERE product_id = ?;";
```
Comment (AnaLanda) [inline src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]:
```suggestion
        String query = "UPDATE product SET isDeleted = true WHERE product_id = ? AND isDeleted = false;";
```

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot delete product with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Product> getAll() {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM product WHERE isDeleted = FALSE;";
```
Comment (AnaLanda) [inline src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]:
```suggestion
            String query = "SELECT * FROM product WHERE isDeleted = false;";
```

File: src/main/java/com/internet/mixshop/util/ConnectionUtil.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class ConnectionUtil {
+    private static final String URL = "jdbc:mysql://localhost:3306/mix?serverTimezone=UTC";
+
+    static {
+        try {
+            Class.forName("com.mysql.jdbc.Driver");
```
Comment (AnaLanda) [inline src/main/java/com/internet/mixshop/util/ConnectionUtil.java]: Intellij Idea says that this is deprecated and we should write `"com.mysql.cj.jdbc.Driver"`

**Review by AnaLanda — COMMENTED** [review]
Looks good overall)

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 1 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 2 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 3 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 4 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/com/internet/mixshop/util/ConnectionUtil.java (hunk 5 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/util/ConnectionUtil.java]: done

File: src/main/java/com/internet/mixshop/Application.java [hunk 6]
```diff
@@ -1,19 +1,20 @@
...
-import com.internet.mixshop.service.ProductService;
 import com.internet.mixshop.service.ShoppingCartService;
 import com.internet.mixshop.service.UserService;
 
 public class Application {
     private static Injector injector = Injector.getInstance("com.internet.mixshop");
 
     public static void main(String[] args) {
-        ProductService productService = (ProductService) injector.getInstance(ProductService.class);
+        final ProductDao productService = (ProductDao)
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/Application.java]: ProductDao productService ?

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+            }
+            return product;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Product " + product.getName() + " is not added!", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> getById(Long id) {
+        String query = "SELECT * FROM product WHERE product_id = ? AND isDeleted = false;";
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]: use either camel case, or snake case everywhere
```suggestion
        String query = "SELECT * FROM product WHERE product_id = ? AND is_deleted = false;";
```

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Product product = getProduct(resultSet);
+                return Optional.ofNullable(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Product with id = " + id + " not found", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]: exception doesn't mean, that there is no product with given id in db
```suggestion
            throw new DataProcessingException("Can't get product with id = " + id, e);
```

Comment (mikhail771) [reply src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java (hunk 7 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/dao/jdbc/ProductDaoJdbcImpl.java]: done

File: src/main/java/com/internet/mixshop/Application.java (hunk 6 above)
Comment (mikhail771) [reply src/main/java/com/internet/mixshop/Application.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**